src/data/transform.ts

```diff
@@ -435,124 +427,159 @@ export const convertToExcalidrawElements = (
   if (!elements) {
     return [];
   }
+
   const elementStore = new ElementStore();
-  // Push all elements to array as there could be cases where element id
-  // is referenced before element is created
-  elements.forEach((element) => {
-    let elementId = element.id || regenerateId(null);
+
+  // ensure unique ids
+  // ---------------------------------------------------------------------------
+  const ids = new Set<ExcalidrawElement["id"]>();
```

Comment:
set is unordered collection so again order won't be preserved here and additionally this will need to be global so other functions can access it as well. I will prefer the older approach of adding it to element store instead and keep it as an array.

Reply:
it was not a good abstraction. Beyond the type mixing as per above, it was also coupling different responsibilities without making it clear what's happening. Let's first untangle all this, and decide after what we want/need to accomplish and decide on a good abstraction.

Right now, the goal of `ids` Set is to ensure id uniqueness, and that's it. The `elementsWithIds` is the original array of elements except they now contain ids.

So far, so good.

Reply:
I am not sure what you mean by different responsibilities, `elementStore` should be treated as a store so I was just adding all elements into it when traversing so order is always maintained